### PR TITLE
[PM-3136] Cipher encryption

### PR DESCRIPTION
**Pull request by dani-garcia**
## Type of change
```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Implement cipher encryption for the mobile SDK

### CODE REVIEW

File: crates/bitwarden/src/mobile/vault/client_ciphers.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+pub struct ClientCiphers<'a> {
+    pub(crate) client: &'a Client,
+}
+
+impl<'a> ClientCiphers<'a> {
+    pub async fn encrypt(&self, req: CipherEncryptRequest) -> Result<CipherEncryptResponse> {
+        let enc = self.client.get_encryption_settings()?;
+
+        let cipher = req.cipher.encrypt(enc, &None)?;
```
Comment (Hinton) [inline crates/bitwarden/src/mobile/vault/client_ciphers.rs:13]: Ciphers can belong to an organization, although they contain the organization_id themselves if they do.

File: crates/bitwarden/src/vault/cipher/cipher.rs [hunk 2]
```diff
@@ -119,3 +123,34 @@ pub struct CipherListView {
...
+            id: self.id,
+            organization_id: self.organization_id,
+            folder_id: self.folder_id,
+            collection_ids: self.collection_ids,
+            name: self.name.encrypt(enc, org_id)?,
+            notes: self.notes.encrypt(enc, org_id)?,
+            r#type: self.r#type,
+            login: self.login.encrypt(enc, org_id)?,
+            identity: self.identity.encrypt(enc, org_id)?,
+            card: self.card.encrypt(enc, org_id)?,
```
Comment (Hinton) [inline crates/bitwarden/src/vault/cipher/cipher.rs:140]: What happens if we get null here?

**Review by Hinton — COMMENTED** [review]
Would be nice with some tests.

File: crates/bitwarden/src/vault/cipher/cipher.rs (hunk 2 above)
Comment (dani-garcia) [reply crates/bitwarden/src/vault/cipher/cipher.rs:140]: Option<T> implements Encryptable and Decryptable if T implements it, and just maps over the optional, so null stays null, and any other values get encrypted or decrypted as usual.

https://github.com/bitwarden/sdk/blob/4ae062e20e6d6bee2abd89af9dda105d19269bdc/crates/bitwarden/src/crypto.rs#L389-L399

**Review by Hinton — APPROVED** [review]

**Pull request merged**